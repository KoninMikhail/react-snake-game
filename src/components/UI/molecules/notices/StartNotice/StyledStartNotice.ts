import styled from 'styled-components';
import { Text } from 'components/UI/atoms/body/Text/Text';
import { up } from 'styled-breakpoints';

export const StyledStartMessageText = styled(Text)`
  font-size: 1em;

  ${up('sm')} {
    font-size: 1.3em;
  }
`;
