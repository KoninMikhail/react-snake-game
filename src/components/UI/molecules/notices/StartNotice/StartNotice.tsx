import { TextAlign } from 'components/UI/atoms/body/TextAlign/TextAlign';
import BlinkAnimation from 'components/UI/atoms/assets/animations/blink-animation';
import { ExtraTextShadow } from '../../../atoms/assets/shadows/ExtraTextShadow';
import { StyledStartMessageText } from './StyledStartNotice';

interface IStartMessageProps {
  align: 'left' | 'right' | 'center';
  content: string;
}

export const StartNotice = ({ align, content }: IStartMessageProps) => (
  <BlinkAnimation duration={1200}>
    <TextAlign data-testid="text-align" align={align}>
      <ExtraTextShadow>
        <StyledStartMessageText as="span">{content}</StyledStartMessageText>
      </ExtraTextShadow>
    </TextAlign>
  </BlinkAnimation>
);
