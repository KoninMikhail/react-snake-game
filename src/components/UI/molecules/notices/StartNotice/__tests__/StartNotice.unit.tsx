import { render, screen } from '@testing-library/react';
import { StartNotice } from '../StartNotice';

describe('Start Notice', () => {
  it('render with properties', () => {
    render(<StartNotice align={'right'} content={'Vasya'} />);

    expect(screen.getByText('Vasya')).toBeInTheDocument();
    expect(screen.getByText('Vasya').tagName.toLowerCase()).toEqual('span');
    expect(screen.getByTestId('text-align')).toHaveStyleRule('text-align', 'right');
  });

  it('render correctly', () => {
    const { container } = render(<StartNotice align={'left'} content={'Vasya'} />);
    expect(container).toMatchSnapshot();
  });
});
