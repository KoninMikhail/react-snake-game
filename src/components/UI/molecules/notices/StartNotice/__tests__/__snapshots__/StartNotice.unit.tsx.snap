// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Notice render correctly 1`] = `
.c1 {
  text-align: left;
}

.c0 {
  -webkit-animation-duration: 1200ms;
  animation-duration: 1200ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-delay: 0;
  animation-delay: 0;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: iDIVWf;
  animation-name: iDIVWf;
  -webkit-animation-timing-function: step-start;
  animation-timing-function: step-start;
}

.c2 {
  text-shadow: 2px 2px 0 rgba(0,0,0,1);
}

.c3 {
  font-size: 1em;
}

@media (min-width:576px) {
  .c3 {
    font-size: 1.3em;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="text-align"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Vasya
        </span>
      </div>
    </div>
  </div>
</div>
`;
